### PR TITLE
Commentable Base class

### PR DESCRIPTION
**Pull request by guynir42**
I added a base class that is the target for all comments and annotations (and in the future maybe other meta-data classes). 

At this point I haven't changed anything outside `models.py` because I want feedback on the implementation. 

There are two main issues to solve:
[ ] how do we migrate all the sub-classes into a new table with new IDs. 
[ ] I changed the `Obj` primary key to be a numeric id and added a `name` column to capture the ZTF name. 

The last point will require a big refactor. I think it is the right thing to do, and it has been the right thing to do from the start. 
It will be painful but it makes more sense to have `Obj` using the same integer identifier as all other classes, and refer to its name by the `name` property. 

Let me know what you think.

### CODE REVIEW

File: skyportal/models.py [hunk 1]
```diff
@@ -547,13 +547,30 @@ def token_groups(self):
...
+class Obj(CommentableBase, ha.Point):
     """A record of an astronomical Object and its metadata, such as position,
     positional uncertainties, name, and redshift."""
 
     update = public
 
-    id = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    # id = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    name = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    id = sa.Column(sa.Integer, sa.ForeignKey('commentablebases.id'), primary_key=True)
```
Comment (acrellin) [inline skyportal/models.py:573]: This will just be inherited from `Base`, like all other `Base`-based models, so no need for this

Comment (acrellin) [reply skyportal/models.py:573]: And I don't think we should be changing `Obj.id` to an int and moving the current `Obj.id` to `Obj.name`

**Review by acrellin — COMMENTED** [review]
I'll take a closer look soon, but this feels overly complicated to me. I think we can use a much simpler class inheritance schema

**Review by dmitryduev — COMMENTED** [review]
@guynir42 so what is your plan on this one?

**Comment by stefanv** [comment]
We may need to live with a compromise for now, since this is a somewhat invasive change.

**Comment by guynir42** [comment]
@stefanv I've moved to a different design using a mixin class. see #1841 
I guess I can close this PR for now.

**Pull request closed without merging**